modules/sdk-core/src/bitgo/wallet/wallet.ts

```diff
@@ -1463,14 +1465,111 @@ export class Wallet implements IWallet {
+      let sharedKeychain: any = undefined;
+
+      if (needsKeychain) {
+        try {
+          const keychain = await this.getEncryptedUserKeychain();
+
+          if (!keychain.encryptedPrv) {
+            sharedKeychain = undefined;
+          } else {
+            const userPrv = decryptKeychainPrivateKey(this.bitgo, keychain, params.walletPassphrase);
+            if (!userPrv) {
+              throw new Error('Unable to decrypt user keychain.');
+            }
+
+            const eckey = makeRandomKey();
+            const sharing = shareOption;
+            assert(sharing.keychain?.toPubKey);
+
+            const secret = getSharedSecret(eckey, Buffer.from(sharing.keychain?.toPubKey, 'hex')).toString('hex');
+            const newEncryptedPrv = this.bitgo.encrypt({ password: secret, input: userPrv });
+
+            let pub = keychain.pub ?? keychain.commonPub;
+            if (keychain.commonKeychain) {
+              pub =
+                this.baseCoin.getMPCAlgorithm() === 'eddsa'
+                  ? EddsaUtils.getPublicKeyFromCommonKeychain(keychain.commonKeychain)
+                  : EcdsaUtils.getPublicKeyFromCommonKeychain(keychain.commonKeychain);
+            }
+
+            sharedKeychain = {
+              pub,
+              encryptedPrv: newEncryptedPrv,
+              fromPubKey: eckey.publicKey.toString('hex'),
+              toPubKey: sharing.keychain?.toPubKey,
+              path: sharing.keychain?.path,
+            };
+          }
+        } catch (e) {
+          if (e.message === 'No encrypted keychains on this wallet.') {
+            sharedKeychain = {};
+          } else {
+            throw e;
+          }
+        }
+        bulkCreateShareOptions.shareOptions.push({
+          user: shareOption.user,
+          permissions: shareOption.permissions,
+          keychain: Object.keys(sharedKeychain ?? {}).length === 0 ? undefined : sharedKeychain,
+        });
+      }
+    }
+    return await this.createBulkShares(bulkCreateShareOptions);
+  }
+
+  /**
+   * Send an encrypted wallet share to BitGo.
+   * @param params
+   */
+  async createBulkShares(params: BulkCreateShareOptions = { shareOptions: [] }): Promise<any> {
+    if (!params.shareOptions || Object.keys(params.shareOptions).length === 0) {
+      throw new Error('shareOptions cannot be empty');
+    }
+    params.shareOptions.forEach((shareOption) => {
+      if (shareOption.keychain) {
+        const { pub, encryptedPrv, fromPubKey, toPubKey, path } = shareOption.keychain;
+        if (!pub || !encryptedPrv || !fromPubKey || !toPubKey || !path) {
+          throw new Error('requires keychain parameters - pub, encryptedPrv, fromPubKey, toPubKey, path');
+        }
+      }
+    });
+    // URL format - `https://bitgo.fakeurl/api/v2/wallet/${this._wallet.id}/walletshares`;
+    const url = `${this.baseCoin.url('/wallet/').replace(this.baseCoin.getChain() + '/', '')}${
+      this._wallet.id
+    }/walletshares`;
```

Comment:
just use `bitGo.url` if you're not using the `coin` url. Or extend the `baseCoin.url` method to replace `coin` with `chain`.

Reply:
The bitgo.url gives the following
```
this.bitgo.url("/hello")
'https://bitgo.fakeurl/api/v1/hello'
```
 While I need a 
` https://bitgo.fakeurl/api/v2/wallet/${this._wallet.id}/walletshares`
I'm using the baseCoin URL and replacing the coin with space(removes the /coin/ part) And adding the walletId

Reply:
this.bitgo.url("/hello", 2) will give you v2 url.